web/srv/api_handlers.go

```diff
@@ -310,3 +320,35 @@ func (h *handler) handleAPIEdges(w http.ResponseWriter, req *http.Request, p htt
 	}
 	renderJSONPb(w, result)
 }
+
+func (h *handler) handleAPICheck(w http.ResponseWriter, req *http.Request, p httprouter.Params) {
+	type CheckResult struct {
+		*healthcheck.CheckResult
+		ErrMsg  string `json:",omitempty"`
+		HintURL string `json:",omitempty"`
+	}
+
+	results := make(map[healthcheck.CategoryID][]*CheckResult)
+
+	collectResults := func(result *healthcheck.CheckResult) {
+		if result.Retry || excludedChecksRE.MatchString(result.Description) {
+			return
```

Comment:
I just realized that the checks that you're skipping here could be failing, and that would cause `RunChecks` to return `false`. But if the only failed checks are the ones that were skipped, that will cause us to display all passing checks in the UI, but with a chip that says that checks are failing. Instead of relying on the return value of `RunChecks`, I think you need to instead calculate success manually inside this function, based only on the checks that are not being skipped.

Reply:
Good point @klingerf, thanks!